### PR TITLE
feat: cff to doi parser

### PR DESCRIPTION
**Pull request by cmdoret**
Adds a parser to extract the DOI from a `CITATION.cff` file into [schema:citation](https://schema.org/citation)

### CODE REVIEW

**Comment by cmdoret** [comment]
With more recent versions of the format, doi can also be represented as:
```yaml
identifiers:
  - description: This is the collection of archived snapshots of all versions of My Research Software
    type: doi
    value: "10.5281/zenodo.123456"
```

* spec: https://github.com/citation-file-format/citation-file-format
* example: https://github.com/opencv/cvat/tree/develop

**Comment by cmdoret** [comment]
We also need to define behaviour in case multiple dois are present in the file [example from the spec repo](https://github.com/citation-file-format/citation-file-format/blob/main/CITATION.cff):

```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.1003149
    description: The concept DOI for the collection containing all versions of the Citation File Format.
  - type: doi
    value: 10.5281/zenodo.5171937
    description: The versioned DOI for the version 1.2.0 of the Citation File Format.
date-released: "2021-08-09"
keywords:
  - citation file format
  - CFF
  - citation files
  - software citation
  - file format
  - YAML
  - software sustainability
  - research software
  - credit
license: "CC-BY-4.0"
doi: 10.5281/zenodo.5171937
```

**Comment by rmfranken** [comment]
>Do we want to generate multiple triples in that case?
yes, I think so. But in the example case you sketch - I would only capture the value for "doi:", and not the values for the identifiers which happen to be DOI's, as they are referring to different abstractions ('is part of', 'specific versions'). If there are multiple values for "doi:" possible for a repository/tool in a CFF, then our tool should reflect that, regardless of how dumb it is to have two identifiers for representing the same "thing".

Note, I see that that the example also contains doi's for conference proceedings etc. I'm referring above to the "doi:" purely at the root of the file, not of objects that the file references as e.g.

`conference:
        name: "Workshop on Sustainable Software for Science: Practice and Experiences (WSSSPE5.1)"
    doi: 10.6084/m9.figshare.3827058
    title: "Track 2 Lightning Talk: Should CITATION files be standardized?"`

Those should not be part of the written rdf output in my opinion. I would not even make them "seeAlso" since they don't seem to be central to what identifies a given repository.

File: gimie/parsers/cff.py
Comment (rmfranken) [inline gimie/parsers/cff.py:1]: Nice! I can understand what you did, and I like the choice of a set over lists. Maybe something we should do elsewhere in our repo, as the order is usually not important, and duplicate triples is also not a thing in RDF, so sets make much more sense in our context!

File: gimie/parsers/__init__.py [hunk 1]
```diff
@@ -72,6 +74,8 @@ def select_parser(
     # Only parse licenses in the root directory
```
Comment (rmfranken) [inline gimie/parsers/__init__.py:74]: suggestion: #only parse licenses & CFF files in the root directory

**Review by rmfranken — CHANGES_REQUESTED** [review]
Code looks nice. Very clear. You gave me a new reason to chase down the parser abstraction, I will need the same kind of logic for my "rdfifyer" script. With the small comment improvement I think this is ready to go!

File: gimie/parsers/__init__.py (hunk 1 above)
Comment (cmdoret) [reply gimie/parsers/__init__.py]: addressed in https://github.com/sdsc-ordes/gimie/pull/107/commits/652a59940a26fa7d946d620d22023d01e88bcd7a

File: gimie/parsers/cff.py
Comment (cmdoret) [reply gimie/parsers/cff.py:1]: happy to help, lmk if you want advice / reviews on the rdfyier! :)

**Comment by cmdoret** [comment]
> > Do we want to generate multiple triples in that case?
> > yes, I think so. But in the example case you sketch - I would only capture the value for "doi:", and not the values for the identifiers which happen to be DOI's, as they are referring to different abstractions ('is part of', 'specific versions'). If there are multiple values for "doi:" possible for a repository/tool in a CFF, then our tool should reflect that, regardless of how dumb it is to have two identifiers for representing the same "thing".
> 
> Note, I see that that the example also contains doi's for conference proceedings etc. I'm referring above to the "doi:" purely at the root of the file, not of objects that the file references as e.g.
> 
> `conference: name: "Workshop on Sustainable Software for Science: Practice and Experiences (WSSSPE5.1)" doi: 10.6084/m9.figshare.3827058 title: "Track 2 Lightning Talk: Should CITATION files be standardized?"`
> 
> Those should not be part of the written rdf output in my opinion. I would not even make them "seeAlso" since they don't seem to be central to what identifies a given repository.

But if we take the other example (https://github.com/opencv/cvat/blob/develop/CITATION.cff), there is no `doi:` at the root, it is only defined through identifier. Looks like we have to choose between missing citations and included unrelated dois. Shall we go with the former and keep things as they are now? (only parse `doi:`)

**Comment by rmfranken** [comment]
Yes, I think the former is simpler & safer. I find that objectifying the doi into some identifier more confusing and hopefully most people just have a single doi for their tool. That being said, we can always change it we see we are missing out on a lot of DOI's...

**Review by rmfranken — APPROVED** [review]

**Pull request merged**